src/main/java/net/coderbot/iris/uniforms/CommonUniforms.java

```diff
@@ -259,22 +258,12 @@ public int getAsInt() {
 
 			ItemStack stack = client.player.getItemInHand(hand);
 
-			if (stack == ItemStack.EMPTY || stack == null) {
+			if (stack == ItemStack.EMPTY || stack == null || !(stack.getItem() instanceof IrisItemLightProvider)) {
 				return 0;
 			}
 
-			if (stack.getItem() instanceof BlockItem) {
-				BlockItem item = (BlockItem)stack.getItem();
-
-				return item.getBlock().defaultBlockState().getLightEmission();
-			}
-			else if (stack.getItem() instanceof IrisItemLightProvider) {
-				IrisItemLightProvider item = (IrisItemLightProvider)stack.getItem();
-
-				return item.getLightEmission(client.player, stack);
-			}
-
-			return 0;
+			IrisItemLightProvider item = (IrisItemLightProvider)stack.getItem();
+			return item.getLightEmission(client.player, stack);
```

Comment:
```suggestion
			return ((IrisItemLightProvider)stack.getItem()).getLightEmission(client.player, stack);
```

Remember to remove the line above!